sdcm/cluster_aws.py

```diff
@@ -443,26 +443,8 @@ def restart(self):
                     for event_filter in event_filters:
                         event_filter.cancel_filter()
 
-    def reboot(self, hard=True, verify_ssh=True):
-        result = self.remoter.run('uptime -s')
-        pre_uptime = result.stdout
-
-        def uptime_changed():
-            result = self.remoter.run('uptime -s', ignore_status=True)
-            return pre_uptime != result.stdout
-
-        if hard:
-            self.log.debug('Hardly rebooting node')
-            self._instance_wait_safe(self._instance.reboot)
-        else:
-            self.log.debug('Softly rebooting node')
-            self.remoter.run('sudo reboot', ignore_status=True)
-
-        # wait until the reboot is executed
-        wait.wait_for(func=uptime_changed, step=3, timeout=180, throw_exc=True)
-
-        if verify_ssh:
-            self.wait_ssh_up()
+    def hard_reboot(self):
```

Comment:
I don't see any reason to move this code out of the `reboot` method, it is the same!

Reply:
It's only relevant for gce and aws, not docker or physical which don't support it.

Reply:
> I don't see any reason to move this code out of the `reboot` method, it is the same!

It's not good to access _instance insider BaseNode.
It's same occasionally, the method name is from the backend lib. For the unsupported backend, we need to raise exception

Reply:
_instance_wait_safe() is need for calling backend async method, but it's only imported in asw & gce backend. So we need to move them out of reboot method, even they look to be same.